woocommerce/wc-admin
# Migrate notes data store to wp data

Fixes #3380

Moves the notes data store from wc-api to wp data under the `data` package. 

* Migrates selectors, actions, and resolvers to the new data store.  Caches all notes under `state.notes` and individual queries reference note IDs from `state.noteQueries`.
* Updates references to these methods in components.
* Makes some updates around checking for requesting/resolving status and note updating status (now stored in individual note).  /cc @octaedro for your review on this piece to make sure things work as expected 😄 
* Moves the notices for dismissal into the components.

### Screenshots
<img width="527" alt="Screen Shot 2020-08-06 at 8 26 25 AM" src="https://user-images.githubusercontent.com/10561050/89531647-3d177f00-d7f9-11ea-8a65-78217c54ea40.png">


### Detailed test instructions:

1. Check all places that retrieve notes (activity panel inbox / historical data / unread indicators / store alerts / profile wizard).
1. Test dismissing individual and all notes.
1. Try to undo both the above dismissals using the undo button in the notice.
1. Make sure notes loading status appears correctly when resolving.
